web/controllers/task_controller.ex

```diff
@@ -57,30 +63,43 @@ defmodule CoursePlanner.TaskController do
   def edit(conn, %{"id" => id}) do
     case Tasks.get(id) do
       {:ok, task} ->
-        render(conn, "edit.html",
-          task: task,
-          changeset: Task.changeset(task),
-          users: Volunteers.all())
+        render(conn, "edit.html", task: task, changeset: Task.changeset(task))
       {:error, :not_found} ->
         conn
         |> put_status(404)
         |> render(CoursePlanner.ErrorView, "404.html")
     end
   end
 
-  def update(conn, %{"id" => id, "task" => params}) do
-    case Tasks.update(id, params) do
-      {:ok, task} ->
-        conn
-        |> put_flash(:info, "Task updated successfully.")
-        |> redirect(to: task_path(conn, :show, task))
-      {:error, :not_found} ->
-        conn
-        |> put_status(404)
-        |> render(CoursePlanner.ErrorView, "404.html")
-      {:error, task, changeset} ->
-        render(conn, "edit.html", task: task, changeset: changeset, users: Volunteers.all())
-    end
+  def update(conn, %{"id" => id, "task" => task_params})do
+    with {:ok, task} <- Tasks.get(id)
+     do
+       changeset =
+         task
+         |> Repo.preload([:volunteers])
+         |> Task.changeset(task_params)
+
+       volunteer_ids = Map.get(task_params, "volunteer_ids", [])
+       volunteers = Repo.all(from v in Volunteers.query(), where: v.id in ^volunteer_ids)
+       changeset = Task.update_volunteer(changeset, :volunteers, volunteers)
```

Comment:
This block of code looks like the same as in `TaskController.create/2`, maybe you can extract them?